src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/DataLayerExpander.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManagerCategoryConnector\Expander;
+
+use FondOfSpryker\Shared\GoogleTagManagerCategoryConnector\GoogleTagManagerCategoryConstants as ModuleConstants;
+
+class DataLayerExpander implements DataLayerExpanderInterface
+{
+    /**
+     * @var \FondOfSpryker\Yves\GoogleTagManagerCategoryConnector\Expander\ProductDataLayerExpanderInterface
+     */
+    protected $productExpander;
+
+    /**
+     * @param \FondOfSpryker\Yves\GoogleTagManagerCategoryConnector\Expander\ProductDataLayerExpanderInterface $productExpander
+     */
+    public function __construct(ProductDataLayerExpanderInterface $productExpander)
+    {
+        $this->productExpander = $productExpander;
```

Comment:
Make this sense? I think the `productExpander` must be an own plugin in the module `fond-of-spryker/google-tag-manager-product-connector`...

Reply:
Mhhhh, not sure about that. we already talked about that and we said that we don't want to do it that way

- dependency between packages
- need to overwrite module dependency provider in pyz
- need to handle different logic based on source for product-propertie (product-detail, category, quote etc.)